src/Tests/Topology/When_configuring_publishers.cs

```diff
@@ -0,0 +1,119 @@
+﻿namespace NServiceBus.Azure.WindowsAzureServiceBus.Tests.Topology
+{
+    using System;
+    using System.Linq;
+    using NServiceBus.AzureServiceBus;
+    using NUnit.Framework;
+
+    [TestFixture]
+    [Category("AzureServiceBus")]
+    public class When_configuring_publishers
+    {
+        private PublishersConfiguration _configuration;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var conventions = new FakeConventions();
+            _configuration = new PublishersConfiguration(conventions);
+        }
+
+        [Test]
+        public void Should_throw_if_mapping_does_not_exist()
+        {
+            var exception = Assert.Throws<InvalidOperationException>(() => _configuration.GetPublishersFor(typeof(MyType)));
+            Assert.AreEqual($"No publishers configured for `{typeof(MyType).FullName}`", exception.Message);
+        }
+
+        [Test]
+        public void Should_not_map_publisher_and_type_if_it_is_not_a_message()
```

Comment:
It shouldn't fail. Simply `publishers configuration` doesn't register mapping if supplied type doesn't represent a message